hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java

```diff
@@ -4077,6 +4077,7 @@ private interface StatisticsAggregator<T> {
       STATS_DATA_CLEANER.
           setName(StatisticsDataReferenceCleaner.class.getName());
       STATS_DATA_CLEANER.setDaemon(true);
+      STATS_DATA_CLEANER.setContextClassLoader(null);
```

Comment:
what are the risks of setting this to null?

Reply:
If the thread relies on the return value of `Thread.currentThread().getContextClassLoader()` for logical processing, it may encounter NPE, but I checked the corresponding code. This risk should not exist. In fact, this is a normal way of handling classloader references in the daemon thread. You can refer https://bugs.java.com/bugdatabase/view_bug.do?bug_id=JDK-7008595